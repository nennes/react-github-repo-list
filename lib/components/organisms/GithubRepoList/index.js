/* @flow */

import axios from "axios";
import React, { PureComponent } from "react";
import { Repo as RepoType, GithubRepo as GithubRepoType } from "../../../types";
import RepoList from "../RepoList";

type Props = {
  username: string
};

type State = {
  lastUpdated?: Date,
  repoList?: Array<RepoType>
};

export default class GithubRepoList extends PureComponent<void, Props, State> {
  constructor(props: Props, context: any) {
    super(props);
    this.state = {};
  }

  state: State;

  componentWillMount() {
    axios
      .get(`https://api.github.com/users/${this.props.username}/repos`, {
        params: {
          sort: "updated"
        }
      })
      .then(response =>
        this.setState({
          lastUpdated: new Date(),
          repoList: this.buildRepoList(response.data)
        })
      )
      .catch(error => console.log(error));
  }

  buildRepoList(apiResponseData: Array<GithubRepoType>): Array<RepoType> {
    console.log(apiResponseData);
    return apiResponseData.map(repo => ({
      name: repo.name,
      title: repo.description,
      url: repo.hmtl_url,
      markdown: "# not implemented yet",
      timestamp: new Date(repo.updated_at)
    }));
  }

  render() {
    if (this.state.repoList) {
      return <RepoList repoList={this.state.repoList} />;
    } else {
      return <div>Loading...</div>;
    }
  }
}
