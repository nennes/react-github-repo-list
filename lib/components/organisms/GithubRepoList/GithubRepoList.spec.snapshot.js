/* global test, expect */

import React from "react";
import GithubRepoList from "./index";
var axios = require("axios");
var MockAdapter = require("axios-mock-adapter");
import renderer from "react-test-renderer";

test("RepoList organism renders correctly", () => {
  var mock = new MockAdapter(axios);

  mock.onGet("https://api.github.com/users/dummy/repos").reply(200, {
    data: [
      {
        markdown: "# Some markdown",
        name: "repository-name",
        timestamp: "2015-12-13T13:36:22Z",
        title: "Repository title",
        url: "http://www.test.com"
      },
      {
        markdown: "# Some more markdown",
        name: "another-repository-name",
        timestamp: "2015-12-15T15:33:22Z",
        title: "Another repository title",
        url: "http://www.example.com"
      }
    ]
  });

  const component = renderer.create(<GithubRepoList username="dummy" />);
  expect(component.toJSON()).toMatchSnapshot();
});
