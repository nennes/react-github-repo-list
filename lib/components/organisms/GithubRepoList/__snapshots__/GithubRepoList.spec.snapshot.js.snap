// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList organism renders correctly 1`] = `
<div>
  Loading...
</div>
`;
